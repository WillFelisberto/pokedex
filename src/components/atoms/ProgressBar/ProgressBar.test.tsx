import { render, screen } from '@/tests/test-utils';

import { ProgressBar } from '.';

describe('<ProgressBar />', () => {
  it('should render gender label', () => {
    const { container } = render(<ProgressBar malePercentage={50} />);

    expect(screen.getByText(/GÃŠNERO/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render male percentage correctly', () => {
    render(<ProgressBar malePercentage={60} />);

    expect(screen.getByText(/60.0%/i)).toBeInTheDocument();
  });

  it('should render female percentage correctly', () => {
    render(<ProgressBar malePercentage={40} />);

    expect(screen.getByText(/60.0%/i)).toBeInTheDocument();
  });

  it('should have the correct styles for progress bars', () => {
    render(<ProgressBar malePercentage={70} />);

    const maleBar = screen.getByTestId('male-bar');
    const femaleBar = screen.getByTestId('female-bar');

    expect(maleBar).toHaveClass('bg-[#2551C3]');
    expect(femaleBar).toHaveClass('bg-[#FF7596]');
  });
});
