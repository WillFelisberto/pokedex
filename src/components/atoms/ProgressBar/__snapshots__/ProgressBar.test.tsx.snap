// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> should render gender label 1`] = `
<div>
  <div
    class="w-full text-center"
  >
    <p
      class="text-gray-700 font-semibold mb-1"
    >
      GÃŠNERO
    </p>
    <div
      class="w-full bg-gray-200 rounded-full h-4 overflow-hidden relative"
    >
      <div
        class="bg-[#2551C3] h-full absolute left-0"
        data-testid="male-bar"
        style="width: 50%;"
      />
      <div
        class="bg-[#FF7596] h-full absolute right-0"
        data-testid="female-bar"
        style="width: 50%;"
      />
    </div>
    <div
      class="flex justify-between text-gray-700 mt-1 text-sm"
    >
      <span
        class="flex items-center font-bold"
      >
        Mars Icon
         
        50.0
        %
      </span>
      <span
        class="flex items-center font-bold"
      >
        Venus Icon
         
        50.0
        %
      </span>
    </div>
  </div>
</div>
`;
